docs(resources.rst): update editor extensions

kanej opened the pull request:
Update the editor extensions section of the resources page:

1. Remove the Atom section - as Atom is being deprecated on 2022-12-15. See https://github.blog/2022-06-08-sunsetting-atom/
2. Update the vim syntax plugins - add the thesis fork of the tomlion vim syntax highlighter. The thesis fork is maintained.
3. add/rename vscode extensions - add truffle and remix extensions and rename the two Solidity extensions to include the authors.

kanej committed:
docs(resources.rst): update editor extensions

Update the editor extensions section of the resources page:

1. Remove the Atom section - as Atom is being deprecated on
   2022-12-15. See https://github.blog/2022-06-08-sunsetting-atom/.
2. Update the vim syntax plugins - remove the tomlion extension as
   no longer maintained, add the thesis and TovarishFin forks of
   the tomlion vim syntax highlighter as these are both maintained.
3. add/rename vscode extensions - add truffle and remix extensions,
   rename the two Solidity extensions to include the authors for
   disambiguation.
4. Clarify the Visual Studio Code title by including `VS Code` in
   brackets, as many people only know it as `VS Code`.
5. Use Sublime Text's full name.

Co-authored-by: franzihei <41991517+franzihei@users.noreply.github.com>
Co-authored-by: Nuno Santos <nunofilipesantos@gmail.com>

cameel commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
+        Syntax highlighting for Solidity in Vim.
+
+    * `Vim Solidity by tomlion <https://github.com/tomlion/vim-solidity/>`_
         Plugin for the Vim editor providing syntax highlighting.
```
I think there's no point keeping the original as a separate recommendation if it's not maintained. I think it would be fine to only mention it in the original point or even remove completely.

cameel commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
 
-    * `Vim Solidity <https://github.com/tomlion/vim-solidity/>`_
+    * `Vim Solidity by Thesis <https://github.com/thesis/vim-solidity/>`_
+        Syntax highlighting for Solidity in Vim.
```
How does it compare to the [fork by TovarishFin](https://github.com/TovarishFin/vim-solidity/commits/master)? They both seem to have been going at it independently for a few years now. Should we list both?

kanej replied:
That makes sense. I will add it in.

cameel commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
+    * `Solidity by Juan Blanco <https://juan.blanco.ws/solidity-contracts-in-visual-studio-code/>`_
         Solidity plugin for Microsoft Visual Studio Code that includes syntax highlighting and the Solidity compiler.
 
+    * `Solidity by Nomic Foundation <https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity>`_
```
It's technically called "Solidity" in the marketplace (probably for better positioning in search results?) but seems clearer to call it Hardhat. Especially when we have Truffle on the same list. [They call it Hardhat themselves](https://hardhat.org/hardhat-vscode).
```suggestion
    * `Hardhat for VS Code by Nomic Foundation <https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity>`_
```

franzihei replied:
Tend to agree with cameel here, might be easier for people to differentiate between Remix, Truffle and Hardhat plugins this way. Unless we strictly want to use the official names from the VS Marketplace.

cameel commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
 
-    * `Hardhat + Solidity Visual Studio Code extension <https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity>`_
-        Solidity and Hardhat support by the Hardhat team.
+    * `Truffle for VS Code <https://marketplace.visualstudio.com/items?itemName=NomicFoundation.hardhat-solidity>`_
```
This says "Truffle" but links to the Hardhat extension.

kanej replied:
Ah, let me get that fixed.

cameel commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
 * Visual Studio Code
 
-    * `Visual Studio Code extension <https://juan.blanco.ws/solidity-contracts-in-visual-studio-code/>`_
+    * `Ethereum Remix <https://marketplace.visualstudio.com/items?itemName=RemixProject.ethereum-remix>`_
+        Ethereum Remix plugins in VSCode
+
+    * `Solidity by Juan Blanco <https://juan.blanco.ws/solidity-contracts-in-visual-studio-code/>`_
```
I think these names look confusing without having something like "extension" or "for VS Code" in them.
```suggestion
    * `Ethereum Remix extension <https://marketplace.visualstudio.com/items?itemName=RemixProject.ethereum-remix>`_
        Ethereum Remix plugins in VSCode

    * `Solidity extension by Juan Blanco <https://juan.blanco.ws/solidity-contracts-in-visual-studio-code/>`_
```

alcuadrado replied:
Maybe the name itself can be in a different format. For example: 

```md
`Solidity` extension by Juan Blanco
```

or 

```md
`Solidity` Visual Studio Code extension, by Juan Blanco
```

cameel replied:
Either of these is fine by me. It's just "Solidity" or "Remix" on its own looked a bit ambiguous to me.

BTW, Not sure if you can put `` ` `` inside the link in RST. At last not in this format.

cameel replied:
By the way, is it Hardhat for VSC or Solidity for VSC? :)

nikola-matic replied:
All of these are in the `Visual Studio Code` section, so adding VSC/VS Code, etc. may be somewhat redundant (to play the devil's advocate).

franzihei replied:
I think I prefer something like

```
Solidity Visual Studio Code extension, by Juan Blanco
```

just because we also call the other extensions by their name (e.g. "Vim Solidity" or "Emacs Solidity" and not only "Solidity"). I think it adds value to keep the Visual Studio Code, even if it's repetitive (since it's all under the VS Code bullet point).

NunoFilipeSantos replied:
Also, **extension pack** is the correct nomenclature on VS Code world.
```
 * `Ethereum Remix, by the Remix Project <https://marketplace.visualstudio.com/items?itemName=RemixProject.ethereum-remix>`_
	        Ethereum Remix extension pack for VS Code
```

kanej replied:
Can I suggest using the contracted `VS Code extension` instead of `Visual Studio Code extension`, so:

```
* Ethereum Remix VS Code extension ...
* Solidity VS Code extension, by Juan Blanco ...
* Solidity VS Code extension, by Nomic Foundation ...
* Solidity Visual Auditor VS Code extension ...
* Truffle for VS Code ...

Leaving Truffle as is because it is close enough?

```

alcuadrado replied:
> extension pack is the correct nomenclature on VS Code world

I think that's one possible category of extensions.

> By the way, is it Hardhat for VSC or Solidity for VSC? :)

We renamed to Solidity, as we are added general support for Solidity, including, for example, in Foundry projects.

cameel replied:
> We renamed to Solidity, as we are added general support for Solidity, including, for example, in Foundry projects.

Looking at it from user's perspective, I think that Hardhat would still be clearer to me on this list, but well, if that's your preferred official naming then I guess we should follow that.

franzihei commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
+        Syntax highlighting for Solidity in Vim.
+
+    * `Vim Solidity by TovarishFin <https://github.com/TovarishFin/vim-solidity>`_
+        Vim syntax file for solidity.
```
```suggestion
        Vim syntax file for Solidity.
```

franzihei commented on docs/resources.rst:
```diff
@@ -71,22 +60,31 @@ Editor Integrations
     * `Vim Syntastic <https://github.com/vim-syntastic/syntastic>`_
         Plugin for the Vim editor providing compile checking.
 
 * Visual Studio Code
```
```suggestion
* Visual Studio Code (VS Code)
```
While we are at it it probably makes sense to add "VS Code" there since most people use that name for Visual Studio Code, especially if we are referencing VS Code in the list below.

NunoFilipeSantos replied:
I agree.

NunoFilipeSantos commented on docs/resources.rst:
```diff
@@ -62,7 +51,7 @@ Editor Integrations
-        Solidity plugin for IntelliJ IDEA (and all other JetBrains IDEs)
+        Solidity plugin for IntelliJ IDEA (and all other JetBrains IDEs).
 
 * Sublime
```
```suggestion
* Sublime Text
```

NunoFilipeSantos reviewed (changes requested):
Left 2 modifications.

cameel reviewed (dismissed):
I think all the points here were addressed so I'm approving but won't merge yet to give @NunoFilipeSantos and @franzihei a chance to have a final look.

cameel commented:
Oh, this also needs commits to be squashed (does not have to be just one but at least review fixes should not be separate commits - they should be fixed up into the original commits they modify).

kanej commented:
> Oh, this also needs commits to be squashed (does not have to be just one but at least review fixes should not be separate commits - they should be fixed up into the original commits they modify).

@cameel I have squashed them down to one commit and updated the text on that commit to reflect the further actions/updates we took.

NunoFilipeSantos reviewed (approved):
Ok to merge
